lib/openhab/core/item_proxy.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require 'delegate'
+require 'forwardable'
+
+module OpenHAB
+  module Core
+    # Class is a proxy to underlying item
+    class ItemProxy < Delegator
+      extend Forwardable
+      def_delegators :__getobj__, :class, :is_a?, :tap, :then, :yield_self, :hash, :kind_of, :to_s
+
+      # Create a new ItemProxy
+      # @param [Item] item to proxy all calls to
+      # Cannot call super here as we want a dynamic object for lookups
+      # rubocop:disable Lint/MissingSuper
+      def initialize(item)
+        # Convert name to java version for faster lookups
+        @item_name = item.name.to_java
+      end
+      # rubocop:enable Lint/MissingSuper
+
+      #
+      # Lookup item from item registry
+      #
+      # rubocop:disable Style/GlobalVars
```

Comment:
You should use single-line disable here (a comment at the end of the line you're disabling). Both because it's simpler, and because this line will leak into the Yard docs